### PR TITLE
stlink: update to 1.5.1.

### PR DESCRIPTION
**Pull request by bra1nwave**

### CODE REVIEW

File: srcpkgs/stlink/template [hunk 1]
```diff
@@ -10,8 +10,8 @@ short_desc="STM32 discovery line Linux programmer"
 maintainer="bra1nwave <brainwave@openmailbox.org>"
 license="3-clause-BSD"
```
Comment (maxice8) [inline srcpkgs/stlink/template]: BSD-3-Clause

File: common/shlibs [hunk 2]
```diff
@@ -2741,7 +2741,7 @@ libgspell-1.so.2 gspell-1.8.0_1
 libotf.so.1 libotf-0.9.16_1
 libimagequant.so.0 libimagequant-2.8.2_1
 libDOtherSide.so.0 dotherside-0.5.2_1
-libstlink-shared.so.1 stlink-1.3.0_1
+libstlink.so.1 stlink-1.5.1_1
```
Comment (maxice8) [inline common/shlibs]: I think it is better to remove this entry since nothing depends on it.

**Comment by maxice8** [comment]
There is also the option to generate manpages with `pandoc` and a gui tool if `gtk+3-devel` is present i will assume you already considered them

**Comment by maxice8** [comment]
On Mon, Sep 17, 2018 at 12:21:16PM -0700, bra1nwave wrote:
> bra1nwave commented on this pull request.
> 
> 
> 
> > @@ -2741,7 +2741,7 @@ libgspell-1.so.2 gspell-1.8.0_1
>  libotf.so.1 libotf-0.9.16_1
>  libimagequant.so.0 libimagequant-2.8.2_1
>  libDOtherSide.so.0 dotherside-0.5.2_1
> -libstlink-shared.so.1 stlink-1.3.0_1
> +libstlink.so.1 stlink-1.5.1_1
> 
> And what if some package will depend on it later? Do I have to add it back?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/2852#discussion_r218195181

If a package depends on it later then it is added back

File: common/shlibs (hunk 2 above)
Comment (maxice8) [reply common/shlibs]: With the discussion on #2750 it has been sorta decided to keep shlibs for packages that still exist on the repo.

Sorry for the incovenience

**Comment by jnbr** [comment]
Upstream disabled cross compilation of the gui: https://github.com/texane/stlink/commit/cc3a5705099019e01dc59f7907d0c5b1b85878d8
I think it should be safe to revert this patch.

**Comment by maxice8** [comment]
ping

**Pull request merged**